### ISSUE TITLE
Running any `vcluster` command can hang the terminal for long periods while user data is sent to `analytics.loft.rocks` making vcluster unusable

### ISSUE THREAD

**Issue by DanHam** [loft-sh/vcluster#1519]
### What happened?

The `vcluster` CLI is configured to always send data to `analytics.loft.rocks` (`34.70.43.141`).

Data is sent to `analytics.loft.rocks`:

* When ever the user executes a command e.g typing `vcluster list` and pressing enter
* When ever the user makes use of shell completion e.g. typing `vcluster l<tab>`

The analytics site can be **very slow** to respond, which means that the `vcluster` command hangs the terminal for long periods. Since `vcluster` does not return control of the terminal to the user until the transaction with `analytics.loft.rocks` completes, the long delays can make `vcluster` practically unusable.

The example capture in the screen shot below shows a transaction taking around 45 seconds to complete.

![Screenshot from 2024-02-05 14-57-48](https://github.com/loft-sh/vcluster/assets/4364798/09b3c018-accb-4179-b95d-b77854a8d8ab)

There is no option to turn off sending data to `analytics.loft.rocks`. I can find no mention of this practice in the documentation.

While I appreciate gathering information on usage can help drive development in the right direction, I firmly believe:

* Sending data should be opt-**_in_** regardless of whether data is anonymized at source
* There should be an option to disable sending data to `analytic.loft.rocks`
* The documentation should clearly state that data is being sent to `analytics.loft.rocks`
* The documentation should make clear what data is being sent and what it is being used for

Ironically, capturing this data is currently making `vcluster` unusable

### What did you expect to happen?

Sending user data to `analytics.loft.rocks` should be opt-in.
Sending data to `analytics.loft.rocks` should not hang the terminal for long periods e.g. 30 - 60 seconds

### How can we reproduce it (as minimally and precisely as possible)?

Use the `vcluster` CLI e.g. run `vcluster --version`

Use wireshark or similar to capture the transaction between the user machine and `analytics.loft.rocks`

### Host cluster Kubernetes version

N/A

### Host cluster Kubernetes distribution

N/A

### vlcuster version

<details>

```console
$ vcluster --version
vcluster version 0.18.1
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

N/A

### OS and Arch

<details>

```
OS: Fedora
Arch: x86_64
```
</details>

**Comment by facchettos** [comment]
Have you tried to use `vcluster telemetry disable` ?

**Comment by facchettos** [comment]
I will look into the issue regarding performances

**Comment by facchettos** [comment]
#1520 should fix the issue that you were facing regarding the performances

**Comment by DanHam** [comment]
> Have you tried to use `vcluster telemetry disable` ?

I'm pretty certain this issue has nothing to do with vcluster telemetry.

> #1520 should fix the issue that you were facing regarding the performances

Thanks. 

So, if I'm reading that correctly, the client will time out after 3 seconds if the telemetry server is down / or has not yet responded. That seems like a fairly sensible compromise. 

However, I was wondering if:

* The time out could be user configurable.
* The choice of whether or not sending analytics data **_at all_** could be exposed to the user.
* If you had taken a look at just why the analytics server has been so slow to respond. 3 seconds is still a long time to wait  :smile: 

@facchettos @FabianKramm 

While #1520 looks to have solved / alleviated the performance / usability issue it does not address the issue of transparency. 

I'm pretty sure that your website does not mention that analytics data will be sent by the vcluster CLI. There is no documentation telling me what data you are collecting and why.

In some ways the issue of trust / transparency is more important than the performance issue. Are there any plans to update the documentation to let users know about this?

Thanks again

**Comment by ThomasK33** [comment]
Hey @DanHam, I appreciate raising your transparency concerns.

> I'm pretty sure that your website does not mention that analytics data will be sent by the vcluster CLI. There is no documentation telling me what data you are collecting and why.

Here are the docs that describe in detail what and why we're collecting analytics data: https://www.vcluster.com/docs/advanced-topics/telemetry.

It has been up there for over six months; you can check this out: https://github.com/loft-sh/vcluster/commits/main/docs/pages/advanced-topics/telemetry.mdx.

A tip for future doc searches would be to type 'analytics' or 'telemetry' in the search bar in the top right of the docs. It's worthwhile.

---

Even though I understand your concerns, we do our best to be transparent about telemetry data.

**Comment by ThomasK33** [comment]
Hey @DanHam, here's a quick update regarding the points you raised here.

- vCluster CLI collecting telemetry data after turning it off is fixed in #1542
- The vclusterctl/vCluster CLI telemetry samples have been included in the docs in #1549
- The analytics service now only accepts HTTPS and redirects any HTTP traffic, so the data transmission is encrypted going forward.

Thanks a lot for raising this; I appreciate the proactiveness.

**Comment by DanHam** [comment]
No problem. Thanks @ThomasK33
